cfme/infrastructure/resource_pool.py

```diff
@@ -5,24 +5,107 @@
 
-from cfme.fixtures import pytest_selenium as sel
-from cfme.web_ui import Quadicon, Region, toolbar as tb, paginator, match_location
-from functools import partial
+from cfme.base.ui import BaseLoggedInPage
+from cfme.exceptions import ResourcePoolNotFound
+from cfme.web_ui import Quadicon, match_location
 from utils.pretty import Pretty
 from utils.providers import get_crud
 from utils.wait import wait_for
 from utils.appliance import Navigatable
 from utils.appliance.implementations.ui import navigator, CFMENavigateStep, navigate_to
+from widgetastic_manageiq import (
+    ItemsToolBarViewSelector, ManageIQTree, PaginationPane, Text, Table, Search, BreadCrumb,
+    SummaryTable, Accordion)
 
 
-details_page = Region(infoblock_type='detail')
+class ResourcePoolToolbar(View):
+    """The toolbar on the main page"""
+    configuration = Dropdown('Configuration')
+    policy = Dropdown('Policy')
+    download = Dropdown('Download')
 
-cfg_btn = partial(tb.select, 'Configuration')
-pol_btn = partial(tb.select, 'Policy')
+    view_selector = View.nested(ItemsToolBarViewSelector)
 
-match_page = partial(match_location, controller='resource_pool',
-                     title='Resource Pools')
+
+class ResourcePoolDetailsToolbar(View):
+    """The toolbar on the details page"""
+    configuration = Dropdown('Configuration')
+    policy = Dropdown('Policy')
+    download = Button(title='Download summary in PDF format')
+
+
+class ResourcePoolDetailsAccordion(View):
+    """The accordian on the details page"""
+    @View.nested
+    class properties(Accordion):  # noqa
+        tree = ManageIQTree()
+
+    @View.nested
+    class relationships(Accordion):  # noqa
+        tree = ManageIQTree()
+
+
+class ResourcePoolEntities(View):
+    """Entities on the main list page"""
+    title = Text('//div[@id="main-content"]//h1')
+    table = Table("//div[@id='list_grid']//table")
+    search = View.nested(Search)
+    # element attributes changed from id to class in upstream-fine+, capture both with locator
+    flash = FlashMessages('.//div[@id="flash_msg_div"]'
+                          '/div[@id="flash_text_div" or contains(@class, "flash_text_div")]')
+
+
+class ResourcePoolDetailsEntities(View):
+    """Entities on the details page"""
+    breadcrumb = BreadCrumb()
+    title = Text('//div[@id="main-content"]//h1')
+    properties = SummaryTable(title='Properties')
+    relationships = SummaryTable(title='Relationships')
+    smart_management = SummaryTable(title='Smart Management')
+
+
+class ResourcePoolView(BaseLoggedInPage):
+    """Base view for header and nav checking, navigatable views should inherit this"""
+    @property
+    def in_resource_pool(self):
+        nav_chain = ['Compute', 'Infrastructure', 'Resource Pools']
+        return (
+            self.logged_in_as_current_user and
+            self.navigation.currently_selected == nav_chain and
+            match_location(controller='resource_pool', title='Resource Pools'))
```

Comment:
Please put TODO here, because `match_location()` should be also converted to widgetastic.